<launch>
    <!-- <arg name="wname" default="second_building"/> -->
    <arg name="wname" default="MultiFloor"/>
    <arg name="rname" default="a2"/>
    <param name="robot_name" type="str" value="$(arg rname)"/>
    <param name="ros_namespace" type="str" value="/$(arg rname)_gazebo/"/>
    <param name="gazebo_model_name" type="str" value="$(arg rname)_gazebo"/>
    <arg name="robot_path" value="(find $(arg rname)_description)"/>
    <arg name="dollar" value="$"/>

    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
    <!-- Debug mode will hung up the robot, use "true" or "false" to switch it. -->
    <arg name="user_debug" default="false"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find rl_sar)/worlds/$(arg wname).world"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)"/>
        <arg name="headless" value="$(arg headless)"/>
    </include>

    <arg name="gpu" default="false"/>
    <node pkg="tf2_ros" type="static_transform_publisher"
      name="map_to_odom"
      args="0 7.5 0 3.1415926 0 0  map odom" />
    <node name="pointcloud_time_adder" pkg="rl_sar" type="addTime.py" output="screen">
    <!-- 参数配置 -->
        <param name="input_topic" value="/velodyne_points" />      <!-- 输入点云话题 -->
        <param name="output_topic" value="/velodyne_points_with_time" />  <!-- 输出点云话题 -->
        <param name="sensor_frame" value="velodyne" />  <!-- 传感器坐标系 -->
        <param name="scan_rate" value="10.0" />
        <param name="clockwise" value="false" />
        <param name="num_lasers" value="16" />  
        <param name="downsample_ratio" value="0.6" />
        <param name="max_points" value="100000" />
        <param name="time_warning_threshold" value="0.1" />
    </node>
    <node name="base_footprint_broadcaster" pkg="rl_sar" type="footprint.py" output="screen" />
    <!-- Load the URDF into the ROS Parameter Server -->
    <param name="robot_description"
           command="$(find xacro)/xacro --inorder '$(arg dollar)$(arg robot_path)/xacro/robot.xacro'
           DEBUG:=$(arg user_debug) gpu:=$(arg gpu)"/>

    <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
    <!-- Set trunk and joint positions at startup -->
    <node pkg="gazebo_ros" type="spawn_model" name="urdf_spawner" respawn="false" output="screen"
          args="-urdf -x 0.0 -y 7.5 -z 1.0 -Y 3.1415926 -model $(arg rname)_gazebo -param robot_description"/>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(arg dollar)$(arg robot_path)/config/robot_control.yaml" command="load"/>

    <!-- convert joint states to TF transforms for rviz, etc -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"
          respawn="false" output="screen">
        <remap from="/joint_states" to="/$(arg rname)_gazebo/joint_states"/>
    </node>

    <!-- <include file="$(find pct_planner)/launch/test.launch">
	</include> -->
    <!-- <include file="$(find fast_lio)/launch/mapping_velodyne.launch" /> -->
    <!-- Load joy node -->
    <node pkg="joy" type="joy_node" name="joy_node" output="screen"/>

</launch>
